sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -1043,6 +1043,171 @@ object functions {
    */
   def var_pop(columnName: String): Column = var_pop(Column(columnName))
 
+  /**
+   * Aggregate function: returns the average of the independent variable for non-null pairs
+   * in a group, where `y` is the dependent variable and `x` is the independent variable.
+   *
+   * @group agg_funcs
+   * @since 3.5.0
+   */
+  def regr_avgx(y: Column, x: Column): Column = withAggregateFunction { RegrAvgX(y.expr, x.expr) }
+
+  /**
+   * Aggregate function: returns the average of the independent variable for non-null pairs
+   * in a group, where `y` is the dependent variable and `x` is the independent variable.
+   *
+   * @group agg_funcs
+   * @since 3.5.0
+   */
+  def regr_avgx(y: String, x: String): Column = regr_avgx(Column(y), Column(x))
```

Comment:
@beliefer I think we don't want to support column name on Scala side.

Reply:
@zhengruifeng could you clarify it? Do we usually only care string version in Python side?

Reply:
for new functions, we should follow https://github.com/apache/spark/blob/56038b5c87508d2211a41b47c192598e42d6c121/sql/core/src/main/scala/org/apache/spark/sql/functions.scala#L59-L61